mungegithub/mungers/README.md

```diff
@@ -0,0 +1,282 @@
+**Approver Selection Mechanism**
+
+# Questions this Doc Seeks To Answer
+
+1. What do we need to get a PR merged into kubernetes/kubernetes?
+
+2. What are reviewers, approvers, and the OWNERS files?
+
+3. How does the reviewer selection mechanism work? approver selection mechanism work?
+
+4. How does an approver know which PR s/he has to approve?
+
+# Overview
+
+Every GitHub directory which is a unit of independent code contains a file named "OWNERS". The file lists reviewers and approvers for the directory. Approvers (or previously called assignees) are owners of the codes. 
+
+Approvers:
+
+* have contributed substantially to the project,
+
+* can provide a final approval (/approved) indicating whether a change to a directory or subdirectory should be accepted
+
+* Approval is done on a per directory basis and subdirectories inherit their parents directories approvers
+
+Reviewers:
+
+* generally a larger set of current and past contributors. 
+
+* They are responsible for a more thorough code review, discussing the implementation details and style
+
+* Provide an /lgtm when they are satisfied with the Pull Request 
+
+An example of the OWNERS file is listed below:
+
+reviewers:
+
+* jack
+
+* ken
+
+* lina
+
+approvers:
+
+* jack
+
+* ken
+
+* lina
+
+# Blunderbuss And Reviewers
+
+### lgtm Label
+
+LGTM is abbreviation for "look good to me". The **lgtm** label is normally given when the code has been thoroughly reviewed.  Getting it means the PR is one step away from getting merged.  Reviewers of the PR give the label to a PR by typing /lgtm in a comment, or retract it by typing /lgtm cancel. Authors of the PR cannot give the label, but they can cancel it. The bot retracts the label automatically if someone updates the PR with a new commit. There is also a bulk LGTM tool that scans multiple short PRs at once, and then gives the label to them in bulk.
+
+Blunderbuss Selection Mechanism
+
+Blunderbuss provides statistical means to select a subset of approvers found in OWNERS files for approving a PR. A PR consists of changes on one or more files, in which each file has different number of lines of codes changed. Blunderbuss determines the magnitude of code change within a PR using total number of lines of codes changed across various files. Number of reviewers selected for each PR is 2.
+
+Algorithm for selecting reviewers is as follows:
+
+(1) determine potential reviewers of a file by going over all reviewers found in the OWNERS files for current and parent directories of the file (deduplication involved)
+
+(2) assign each changed file with a weightage based on number of lines of codes changed
+
+(3) assign each potential reviewer with a weightage by summing up weightages of all changed files in which s/he is a reviewer
+
+(4) randomly select 2 reviewers
+
```

Comment:
This is inconsistent with what you describe above.  Make sure to mention that it's randomly selected according to the weighting